blocklint/main.py

```diff
@@ -38,20 +38,47 @@ def main(args=None):
             with open(file, 'r') as handle:
                 total_issues += process_file(handle, file, word_checkers,
                                              args['end_pos'])
-
     if (args['max_issue_threshold'] is not None
             and args['max_issue_threshold'] <= total_issues):
         print(("Found {issues} issues, with maximum set to "
                "{max}!").format(
                    issues=total_issues,
                    max=args['max_issue_threshold']))
         sys.exit(1)
+        
+
+def clean_ignored_docstrings(lines):
+    end_pattern = re.compile(r'("""\s*#\s*blocklint:.*pragma$|\'\'\'\s*#\s*blocklint:.*pragma$)')
+    lines_to_clear = []
+
+    for i, line in enumerate(lines):
+        match = end_pattern.search(line)
+        if match:
+            start_quote = match.group(1)[:3]  # This is either """ or '''
+            
+            if start_quote in line[:-(len(match.group(1)))]:
+                # single line docstring
+                lines_to_clear.append((i, i+1))
+            else:
+                # multi line docstring - scan backwards from the current line
+                for j in range(i-1, -1, -1):
```

Comment:
I'm curious what would happen in a badly formatted doc string, like it's missing the starting quote.  I don't think it's worth handling well, just that it doesn't explode.

Reply:
2 cases I can think of if a closing triplequote + pragma line is found that doesn't have a matching opening triplequote:
 - It hits the first line of the file - easy enough to check that and raise.
 - It hits a triplequote that belongs to a previous complete docstring - I think it would just ignore all the lines in between. It wouldn't explode, though, just silently ignore more than was intended.
 
 I'll mess around with it when I have some time. Thanks for your feedback!